Cleanup and add details (#93)
0xProject/wiki: tutorials/market-making-on-0x.md

#### What about order matching?

There are two dominant relayer strategies in existence today: [matching](https://0xproject.com/wiki#Matching) and [open orderbook](https://0xproject.com/wiki#Open-Orderbook). Open orderbook relayers host orders that anyone can fill by directly sending a transaction (with the order details and desired fill amount) to the Ethereum blockchain. Using this model, the order in which transactions are filled is determined by the same algorithm used by miners to decide which transactions to include in the next block (more on this later).

Matching relayers require traders to create the inverse 0x order to the one they wish to fill and submit it back to the matcher. The matching relayer will then fill both orders in one atomic transaction to Ethereum on the trader's behalf.

In both models, the relayer never has custody over a trader's assets.

### Creating orders

> 0x orders only exist off-chain and are completely free to create. In order to create a valid 0x order, you can use the [@0x/order-utils](https://0xproject.com/docs/order-utils) Typescript/Javascript library, or alternatively the [0x-order-utils.py](http://0x-order-utils-py.s3-website-us-east-1.amazonaws.com/) Python library. There libraries will help you:

Reviewer: ```suggestion
0x orders only exist off-chain and are completely free to create. In order to create a valid 0x order, you can use the [@0x/order-utils](https://0xproject.com/docs/order-utils) Typescript/Javascript library, or alternatively the [0x-order-utils.py](http://0x-order-utils-py.s3-website-us-east-1.amazonaws.com/) Python library. These libraries will help you:
```

If you want to be notified whenever the order is filled by a trader, you should also submit the order to an [OrderWatcher](https://0xproject.com/docs/order-watcher) instance you are running.

### Cancelling orders

There are multiple ways to cancel 0x orders, each of which is described in the [cancelling orders](https://github.com/0xProject/0x-protocol-specification/blob/master/v2/v2-specification.md#cancelling-orders) section of the protocol specification.

The `cancelOrdersUpTo` approach is perhaps the least straight forward but also the most powerful for a market maker. It allows for the cancellation of an arbitrary number of orders for a fixed amount of gas. By creating orders where the `salt` field is equal to the current unix timestamp in milliseconds, you can cancel all orders that were created at or before a certain time in a single `cancelOrdersUpTo` call (a fixed size transaction). Note that any future orders created with a `salt` that is below the largest `salt` argument of a `cancelOrdersUpTo` call by your address will automatically be invalid.

> Explicitly cancelling orders always requires an on-chain transaction. However, you may create short lived orders and replace the expired orders completely off-chain. Due to the variability of transaction inclusion times in blocks, it it not recommended to set extremely agressive expiration times for orders (empirically, most market makers set order expirations to ~5 minutes).

Reviewer: ```suggestion
Explicitly cancelling orders always requires an on-chain transaction. However, you may create short lived orders and replace the expired orders completely off-chain. Due to the variability of transaction inclusion times in blocks, it is not recommended to set extremely agressive expiration times for orders (empirically, most market makers set order expirations to ~5 minutes).
```

### Developer tooling

#### What is the state of 0x developer tooling?

Currently we have the best tooling support for Javascript/Typescript and are actively improving our support for Python. Visit the [Developers section](https://0xproject.com/docs) of our site for a full-list of developer tools avaiable.

If you plan on implementing custom infrastructure, you will need the following functionality at a bare minimum:

> 1. Interacting with relayers over Websockets or HTTP

Reviewer: ```suggestion
1. Interacting with relayers over Websockets or HTTPS
```

1. Encoding the [`assetData`](https://github.com/0xProject/0x-protocol-specification/blob/master/v2/v2-specification.md#assetdata) of an order using [ABIv2](https://solidity.readthedocs.io/en/latest/abi-spec.html)
1. Hashing orders
1. Cryptographically signing orders with the address you wish to trade with

In addition, it is highly recommended to be connected to an Ethereum node so that you can:

1. Read the state of orders

> 1. Be notified when the state of an order is changed (by subscribing to the relevant events)

Reviewer: ```suggestion
1. Be notified when the state of an order changes (by subscribing to the relevant events)
```
